porch/pkg/git/git.go

```diff
@@ -486,7 +506,27 @@ func (r *gitRepository) loadPackageRevision(ctx context.Context, version, path s
 
 	var ref *plumbing.Reference = nil // Cannot determine ref; this package will be considered final (immutable).
 
-	packageRevision, err := krmPackage.buildGitPackageRevision(ctx, version, ref)
+	var revision string
+	var description string
+	last := strings.LastIndex(version, "/")
+
+	if strings.HasPrefix(version, "drafts/") || strings.HasPrefix(version, "proposed/") {
+		// the passed in version is a ref to an unpublished package revision
+		description = version[last+1:]
+	} else {
+		// the passed in version is a ref to a published package revision
+		if last < 0 {
```

Comment:
How do we handle packagerevisions on the main branch here? Those will not follow the naming convention of tags.